purchase_invoice_plan/models/purchase.py

```diff
@@ -0,0 +1,334 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_compare, float_round
+
+
+class PurchaseOrder(models.Model):
```

Comment:
please, split this file.
We should have `purchase_order.py` and `purchase_invoice_plan.py`.

Reply:
A reminder for next time: the split should be done before doing the migration because it makes easier to fwd/backport changes :wink: